Update on the Lc database

DelloStritto opened the pull request:
1) Splitting the pT bin 4-8 in 4-6 and 6-8 in the skimming part.
2) Change the MC path to the dedicated one

benedikt-voelkel commented on machine_learning_hep/data/data_prod_20200304/database_ml_parameters_LcpK0spp_20200301.yml:
```diff
@@ -212,49 +215,53 @@ LcpK0spp:
-                                 /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2018_data/376_20200304-2028/pklskml] #list of periods
-      pkl_skimmed_merge_for_ml_all: /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_data_mltot
-      pkl_evtcounter_all: /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_data_evttot
+      pkl: [/data/DerivedLuigi/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2016_data/374_20200304-2028/pkl,
```
Can we call these differently? Having a name inside might make people think that's a private production and hence it might be deleted. 
I suggest something like `/data/Derived/LckAnywithJets/.../<final_folder>_updated_ML`
or what do you think @ginnocen @DelloStritto ?

benedikt-voelkel replied:
There are also a lot of other directories...

DelloStritto commented:
Ciao Benedikt, 
My idea was to test these set of skimming data for the jets and then (if evrything it's fine), to delete the actual production (/data/Derived) and to move the /data/DerivedLuigi in data/Derived. 



Da: "benedikt-voelkel" <notifications@github.com> 
A: "ginnocen" <MachineLearningHEP@noreply.github.com> 
Cc: "DelloStritto" <luigi.dellostritto@sa.infn.it>, "Mention" <mention@noreply.github.com> 
Inviato: Venerdì, 20 marzo 2020 14:18:18 
Oggetto: Re: [ginnocen/MachineLearningHEP] Update on the Lc database (#594) 





@benedikt-voelkel commented on this pull request. 


In [ https://github.com/ginnocen/MachineLearningHEP/pull/594#discussion_r395628993 | machine_learning_hep/data/data_prod_20200304/database_ml_parameters_LcpK0spp_20200301.yml ] : 
>        seedmerge: [12,12,12] #list of periods
       period: [LHC16pp,LHC17pp,LHC18pp] #list of periods
       unmerged_tree_dir: [/data/TTree/D0DsLckAnywithJets/vAN-20200304_ROOT6-1/pp_2016_data/374_20200304-2028/merged,
                           /data/TTree/D0DsLckAnywithJets/vAN-20200304_ROOT6-1/pp_2017_data/375_20200304-2028/merged,
                           /data/TTree/D0DsLckAnywithJets/vAN-20200304_ROOT6-1/pp_2018_data/376_20200304-2028/merged] #list of periods
-      pkl: [/data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2016_data/374_20200304-2028/pkl,
-            /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2017_data/375_20200304-2028/pkl,
-            /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2018_data/376_20200304-2028/pkl] #list of periods
-      pkl_skimmed: [/data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2016_data/374_20200304-2028/pklsk,
-                    /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2017_data/375_20200304-2028/pklsk,
-                    /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2018_data/376_20200304-2028/pklsk] #list of periods
-      pkl_skimmed_merge_for_ml: [/data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2016_data/374_20200304-2028/pklskml,
-                                 /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2017_data/375_20200304-2028/pklskml,
-                                 /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2018_data/376_20200304-2028/pklskml] #list of periods
-      pkl_skimmed_merge_for_ml_all: /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_data_mltot
-      pkl_evtcounter_all: /data/Derived/LckAnywithJets/vAN-20200304_ROOT6-1/pp_data_evttot
+      pkl: [/data/DerivedLuigi/LckAnywithJets/vAN-20200304_ROOT6-1/pp_2016_data/374_20200304-2028/pkl, 


There are also a lot of other directories... 

— 
You are receiving this because you were mentioned. 
Reply to this email directly, [ https://github.com/ginnocen/MachineLearningHEP/pull/594#discussion_r395628993 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/ALHMJZSMMPQ6PAIBNA3PUGTRINUJVANCNFSM4LQLLXMQ | unsubscribe ] .

benedikt-voelkel commented:
Ciao @DelloStritto 
Sounds good, let's do that.

@vkucera @nzardosh can you check whether your jet analyses would still work with this PR, or - if not - explain potential show stoppers?! 

Depending on that we would either merge this PR or modify it to introduce 2 DBs, one for jets and one for mult. analyses.
Thanks :)

vkucera commented:
Hi @benedikt-voelkel ,
I don't see anything that would break the jet analysis.

benedikt-voelkel commented:
Hello @ginnocen 
It seems that PR is fine and @vkucera doesn't spot anything incompatible with the jet analyses.

@DelloStritto could you change this PR to have the generic path names everywhere? Then we can merge it, I would say.

Aftwerwards, I would propose: We move the current production from `/data/Derived/LckAnywithJets` to `/data/Derived/LckAnywithJets_bak_<current_date>`. At the same time, we move `/data/DerivedLuigi/LckAnywithJets` to `/data/Derived/LckAnywithJets` and make sure it's everything consistent with the DB. If we don't spot any bugs till Mondaye evening, we delete the backup?!

Any objections?

DelloStritto commented:
Seems to be a good plan. 



Da: "benedikt-voelkel" <notifications@github.com> 
A: "ginnocen" <MachineLearningHEP@noreply.github.com> 
Cc: "DelloStritto" <luigi.dellostritto@sa.infn.it>, "Mention" <mention@noreply.github.com> 
Inviato: Venerdì, 20 marzo 2020 18:22:54 
Oggetto: Re: [ginnocen/MachineLearningHEP] Update on the Lc database (#594) 





Hello [ https://github.com/ginnocen | @ginnocen ] 
It seems that PR is fine and [ https://github.com/vkucera | @vkucera ] doesn't spot anything incompatible with the jet analyses. 

[ https://github.com/DelloStritto | @DelloStritto ] could you change this PR to have the generic path names everywhere? Then we can merge it, I would say. 

Aftwerwards, I would propose: We move the current production from /data/Derived/LckAnywithJets to /data/Derived/LckAnywithJets_bak_<current_date> . At the same time, we move /data/DerivedLuigi/LckAnywithJets to /data/Derived/LckAnywithJets and make sure it's everything consistent with the DB. If we don't spot any bugs till Mondaye evening, we delete the backup?! 

Any objections? 

— 
You are receiving this because you were mentioned. 
Reply to this email directly, [ https://github.com/ginnocen/MachineLearningHEP/pull/594#issuecomment-601819390 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/ALHMJZSLXHI3SLATGRORDELRIOQ65ANCNFSM4LQLLXMQ | unsubscribe ] .

DelloStritto commented:
I updated the pull request and I moved the new skimmed data in /data/Derived. 
I renamed the old one LckAnywithJets to LckAnywithJets_oldskimming. You can rename it as you want. 
Let me know if everything is ok. 



Da: "benedikt-voelkel" <notifications@github.com> 
A: "ginnocen" <MachineLearningHEP@noreply.github.com> 
Cc: "DelloStritto" <luigi.dellostritto@sa.infn.it>, "Mention" <mention@noreply.github.com> 
Inviato: Venerdì, 20 marzo 2020 18:22:54 
Oggetto: Re: [ginnocen/MachineLearningHEP] Update on the Lc database (#594) 





Hello [ https://github.com/ginnocen | @ginnocen ] 
It seems that PR is fine and [ https://github.com/vkucera | @vkucera ] doesn't spot anything incompatible with the jet analyses. 

[ https://github.com/DelloStritto | @DelloStritto ] could you change this PR to have the generic path names everywhere? Then we can merge it, I would say. 

Aftwerwards, I would propose: We move the current production from /data/Derived/LckAnywithJets to /data/Derived/LckAnywithJets_bak_<current_date> . At the same time, we move /data/DerivedLuigi/LckAnywithJets to /data/Derived/LckAnywithJets and make sure it's everything consistent with the DB. If we don't spot any bugs till Mondaye evening, we delete the backup?! 

Any objections? 

— 
You are receiving this because you were mentioned. 
Reply to this email directly, [ https://github.com/ginnocen/MachineLearningHEP/pull/594#issuecomment-601819390 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/ALHMJZSLXHI3SLATGRORDELRIOQ65ANCNFSM4LQLLXMQ | unsubscribe ] .

benedikt-voelkel commented:
Looks good to me @DelloStritto 

@ginnocen can you merge this PR?

ginnocen commented:
@DelloStritto very good! it looks all good for me.  @vkucera @benedikt-voelkel could you also have a look? Vit, i would check in particular if the new productions look good in terms of jet response matrix. If I remember correctly in the past we were observing strange things related to that. Thanks a lot, GM

benedikt-voelkel commented:
Ciao @ginnocen @DelloStritto @vkucera 
I ran the multiplicity analysis vs. V0M percentile and the results look good. I vote for merging this PR :)

vkucera commented on machine_learning_hep/data/data_prod_20200304/database_ml_parameters_LcpK0spp_20200301.yml:
```diff
@@ -212,49 +215,53 @@ LcpK0spp:
-      fracmerge: [0.1,0.1,0.1] #list of periods
+      fracmerge: [[0.1,0.1,0.1,0.4,1.0,1.0],
+                  [0.1,0.1,0.1,0.4,1.0,1.0],
+                  [0.1,0.1,0.1,0.4,1.0,1.0]]
```
This requires the changes in #575 in order to work.

vkucera replied:
Actually, I suspect the changes suggested in #575 might not make it work either because they make `filesel` pt dependent but then `filesel` is used later outside the `for` loop to merge `list_sel_evt` and `list_sel_evtorig` which then should be done also in pt bins I guess.

vkucera commented on machine_learning_hep/data/data_prod_20200304/database_ml_parameters_LcpK0spp_20200301.yml:
```diff
@@ -234,27 +239,29 @@ LcpK0spp:
-      fracmerge: [1.0,1.0,1.0] #list of periods
+      fracmerge: [[1.0,1.0,1.0,1.0,1.0,1.0],
+                  [1.0,1.0,1.0,1.0,1.0,1.0],
+                  [1.0,1.0,1.0,1.0,1.0,1.0]] #list of periods
```
same here
